### PR TITLE
PMM-6693 Endpoint to get current API Key

### PR DESCRIPTION
**Pull request by BupycHuk**
https://github.com/Percona-Lab/pmm-submodules/pull/1479

### CODE REVIEW

**Comment by atymchuk** [comment]
@BupycHuk Can we have a FB to test it?

**Comment by BupycHuk** [comment]
@atymchuk https://github.com/Percona-Lab/pmm-submodules/pull/1479

File: pkg/api/api.go [hunk 1]
```diff
@@ -224,6 +224,10 @@ func (hs *HTTPServer) registerRoutes() {
 			orgsRoute.Get("/", Wrap(GetOrgByName))
 		}, reqGrafanaAdmin)
 
+		// Current user role
+		apiRoute.Group("/auth/key", func(keyRoute routing.RouteRegister) {
```
Comment (atymchuk) [inline pkg/api/api.go:228]: I guess it should be `/auth/role`, since this is what we return.

**Review by atymchuk — CHANGES_REQUESTED** [review]

File: pkg/api/api.go (hunk 1 above)
Comment (BupycHuk) [reply pkg/api/api.go:228]: Actually we return `ApiKeyDTO`, role is only one field of it. I used the same convention which is used in other API endpoints.

**Review by atymchuk — CHANGES_REQUESTED** [review]

File: pkg/api/apikey.go [hunk 2]
```diff
@@ -9,6 +9,27 @@ import (
 	"github.com/grafana/grafana/pkg/models"
 )
 
+func GetAPIKeyCurrent(c *models.ReqContext) Response {
```
[the remark this answers is not in the document]
Comment (BupycHuk) [reply pkg/api/apikey.go:12]: I used the same convention which is used in other API endpoints. like `GetOrgCurrent`

File: pkg/api/api.go [hunk 3]
```diff
@@ -224,6 +224,10 @@ func (hs *HTTPServer) registerRoutes() {
 			orgsRoute.Get("/", Wrap(GetOrgByName))
 		}, reqGrafanaAdmin)
 
+		// Current user role
```
[the remark this answers is not in the document]
Comment (atymchuk) [reply pkg/api/api.go:227]: Ok, nevermind )

**Review by atymchuk — APPROVED** [review]

File: pkg/api/apikey.go [hunk 4]
```diff
@@ -9,6 +9,28 @@ import (
...
+	query := models.GetApiKeyByIdQuery{ApiKeyId: c.ApiKeyId}
+
+	if err := bus.Dispatch(&query); err != nil {
+		return Error(500, "Failed to list api keys", err)
+	}
+
+	var expiration *time.Time = nil
+	if query.Result.Expires != nil {
+		v := time.Unix(*query.Result.Expires, 0)
+		expiration = &v
```
Comment (askomorokhov) [inline pkg/api/apikey.go:22]: Can this be done in one line? :)

**Review by askomorokhov — APPROVED** [review]

File: pkg/api/apikey.go (hunk 4 above)
Comment (atymchuk) [reply pkg/api/apikey.go:22]: @askomorokhov it's a nit ))

**Review by atymchuk — APPROVED** [review]

File: pkg/api/apikey.go (hunk 4 above)
Comment (BupycHuk) [reply pkg/api/apikey.go:22]: Unfortunately no, time.Unix is a function

Comment (AlekSi) [reply pkg/api/apikey.go:22]:
```go
expiration = pointer.ToTime(time.Unix(*query.Result.Expires, 0))
```

https://pkg.go.dev/github.com/AlekSi/pointer#ToTime

Comment (BupycHuk) [reply pkg/api/apikey.go:22]: 😊 code won't be consistent :)
Then I'll have to replace it everywhere

Comment (AlekSi) [reply pkg/api/apikey.go:22]: Probably. Just pointing out that it _can_ be; not necessary _should_ be.

**Pull request merged**